[Step5, 6] PR보냅니다.

jaeyeon93 opened the pull request:
PR을 안눌렀네요.. 다시보냅니다.

jaeyeon93 committed:
step5 start

jaeyeon93 committed:
상속을 이용해서 Triangle 동작,

jaeyeon93 committed:
Step5 finish

jaeyeon93 committed:
step5 square, triangle 동작완료, 상속의 super를 이용해서

jaeyeon93 committed:
step6 요구사항 : ArrayList를 사용하는 곳이 있다면 다음과 같이 List 인터페이스를 적용

jaeyeon93 committed:
[Step5,6] 코드수정완료

javajigi commented on src/main/java/Draw.java:
```diff
@@ -2,7 +2,7 @@
 public class Draw {
-    private ArrayList<Point> list = new ArrayList<>();
+    private List<Point> list = new ArrayList<>();
     public Draw(List<String> points){
```
인스턴스 변수와 생성자 사이에 빈 공백 라인 추가..

javajigi commented on src/main/java/Draw.java:
```diff
@@ -2,7 +2,7 @@
-    private ArrayList<Point> list = new ArrayList<>();
+    private List<Point> list = new ArrayList<>();
     public Draw(List<String> points){
         for(int i = 0; i < points.size(); i++){
```
foreach 구문 사용.

javajigi commented on src/main/java/Figure.java:
```diff
@@ -0,0 +1,23 @@
+        return Math.pow((Math.pow(first.xValue() - second.xValue(), 2) + Math.pow(first.yValue() - second.yValue(), 2)) , 0.5);
+    }
+
+    public abstract double area();
```
이 메소드를 위한 인터페이스를 추가하고 이 메소드를 인터페이스에서 구현해 본다.

javajigi commented on src/main/java/Line.java:
```diff
@@ -1,25 +1,14 @@
     }
 
-
+    double distance = super.distance(getPoint(1), getPoint(0));
```
이와 같이 구현하지 말고 메소드로 구현한다.

javajigi commented on src/main/java/Main.java:
```diff
@@ -6,6 +6,7 @@ public static void main (String [] args){
+
         int numberOfPoints = points.size();
 
         if(points.size() == 2) {
```
if 2,3,4에 따라 각기 다른 Figure를 생성하는 부분을 제거해 본다.

javajigi commented on src/main/java/Triangle.java:
```diff
@@ -1,33 +1,17 @@
-    }
+    double a = super.distance(getPoint(0), getPoint(1));
+    double b = super.distance(getPoint(1), getPoint(2));
+    double c = super.distance(getPoint(2), getPoint(0));
```
메소드로 구현해 본다.

javajigi reviewed (changes requested):
구현 잘 했다. 피드백 몇 개 남겼다.

jaeyeon93 committed:
2018-3-8 coordinate finish

jaeyeon93 committed:
area 인터페이스화, ResultView 바꾸기전

jaeyeon93 committed:
피드백 완료

jaeyeon93 committed:
코드충돌해결
Merge branch 'step5' into step5-m

# Conflicts:
#	src/main/java/Line.java
#	src/main/java/Main.java
#	src/main/java/ResultView.java

jaeyeon93 committed:
Move resultview

javajigi commented on src/main/java/Figure.java:
```diff
@@ -18,6 +17,8 @@ public Point getPoint(int i) {
     public double distance(Point first, Point second) {
         return Math.pow((Math.pow(first.xValue() - second.xValue(), 2) + Math.pow(first.yValue() - second.yValue(), 2)) , 0.5);
```
이 메소드를 Line 객체를 이용해 구현할 수는 없을까?

javajigi commented on src/main/java/Square.java:
```diff
@@ -1,14 +1,19 @@
+        return (int)super.distance(getPoint(start), getPoint(end));
+    }
+
+    int width = distance(0, 1);
```
0, 1 값을 하드코딩하기보다 상수로 이름을 부여하면 더 좋겠다.

javajigi reviewed (changes requested):
마지막 피드백이 되기를..

jaeyeon93 committed:
피드백 되돌리기전

javajigi commented on src/main/java/Square.java:
```diff
@@ -4,17 +4,19 @@
-    int width = distance(0, 1);
-    int height = distance(1, 2);
+    public int height() {
+        return (int)line.distance(super.getPoint(1), super.getPoint(2));
```
여기서 Line을 사용하는 것이 아니라 Figure의 distance에서 Line을 사용해 보라는 의미였음.

javajigi commented on src/main/java/Line.java:
```diff
@@ -5,5 +5,10 @@ public Line(List<String> points) {
 
-    double distance = super.distance(getPoint(1), getPoint(0));
+    public double distance(Point p1, Point p2) {
+        return Math.pow((Math.pow(p1.xValue() - p2.xValue(), 2) + Math.pow(p1.xValue() - p2.yValue(), 2)), 0.5);
```
`p1.distance(p2)`와 같이 기존의 Point의 distance를 유지하는 것도 좋지 않을까?

javajigi reviewed (changes requested):
피드백 남겼음.

jaeyeon93 committed:
코드리뷰 수정
Point의 distance를 이용해서 길이 변환 수정

jaeyeon93 committed:
삼각형 넓이 오류 수정
